### PR TITLE
PWN-5745 - Fix crashes

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5745
https://p2pvalidator.atlassian.net/browse/PWN-5771
https://p2pvalidator.atlassian.net/browse/PWN-5949

### CODE REVIEW

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/MainFragment.kt [hunk 1]
```diff
@@ -53,7 +53,7 @@ class MainFragment : BaseFragment(R.layout.fragment_main), MainTabsSwitcher, Cen
                 .withArgs(ARG_MAIN_FRAGMENT_ACTIONS to actions)
     }
 
-    private var onCreateActions: ArrayList<MainFragmentOnCreateAction> by args(
+    private var onCreateActions: ArrayList<MainFragmentOnCreateAction>? by args(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/MainFragment.kt:56]: can't be null - default value is provided

File: app/src/main/java/org/p2p/wallet/moonpay/ui/new/NewBuyPresenter.kt [hunk 2]
```diff
@@ -204,7 +205,10 @@ class NewBuyPresenter(
 
     private fun isValidCurrencyForPay(): Boolean {
         val selectedCurrencyCode = selectedCurrency.code
-        if (selectedPaymentMethod.method == PaymentMethod.MethodType.BANK_TRANSFER) {
+        if (selectedPaymentMethod == null) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/new/NewBuyPresenter.kt:208]: you can do it this check with local variable to remove `?:` later

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/moonpay/ui/new/NewBuyPresenter.kt (hunk 2 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/moonpay/ui/new/NewBuyPresenter.kt:208]: Yeah, but i don't understand with which value you want use ?:

**Review by 1sakov — APPROVED** [review]

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/showKeyboard.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+fun View.hideKeyboard() = (this.context as? Activity)?.hideKeyboard()
+
+fun Fragment.showKeyboard() = activity?.let(FragmentActivity::showKeyboard)
+fun Fragment.hideKeyboard() = activity?.hideKeyboard()
+
+fun Context.showKeyboard() = (this as? Activity)?.showKeyboard()
+fun Context.hideKeyboard() = (this as? Activity)?.hideKeyboard()
+
+fun Activity.showKeyboard() = WindowInsetsControllerCompat(window, window.decorView).show(WindowInsetsCompat.Type.ime())
+fun Activity.hideKeyboard() = WindowInsetsControllerCompat(window, window.decorView).hide(WindowInsetsCompat.Type.ime())
```
Comment (1sakov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/showKeyboard.kt:21]: Please name it correctly. Smth like `KeyboardUtils`

File: app/src/main/java/org/p2p/wallet/moonpay/ui/new/NewBuyFragment.kt [hunk 4]
```diff
@@ -170,12 +163,13 @@ class NewBuyFragment :
...
             preselectedCurrency = selectedCurrency,
             currencies = currencies,
-            requestKey = KEY_REQUEST_CURRENCY,
+            requestKey = KEY_REQUEST,
             resultKey = KEY_RESULT_CURRENCY
         )
     }
 
     override fun setCurrencyCode(selectedCurrencyCode: String) {
+        Timber.tag("_______selectedCurrencyCode").d("$selectedCurrencyCode, view = ${binding.amountsView}")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/new/NewBuyFragment.kt:172]: revert?

File: app/src/main/java/org/p2p/wallet/moonpay/ui/new/NewBuyPresenter.kt [hunk 5]
```diff
@@ -266,12 +270,14 @@ class NewBuyPresenter(
...
             val baseCurrencyCode = selectedCurrency.code.lowercase()
+            val paymentMethod = selectedPaymentMethod?.paymentType ?: return@launch
+
             val result = moonpayBuyInteractor.getMoonpayBuyResult(
                 baseCurrencyAmount = amountInCurrency,
                 quoteCurrencyAmount = amountInTokens,
                 tokenToBuy = selectedToken,
                 baseCurrencyCode = baseCurrencyCode,
-                paymentMethod = selectedPaymentMethod.paymentType
+                paymentMethod = paymentMethod
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/new/NewBuyPresenter.kt:280]: payment method is not null, you can remove ? in the parameter type

File: app/src/main/res/layout/fragment_new_buy.xml [hunk 6]
```diff
@@ -14,136 +14,146 @@
...
+                    android:layout_marginStart="24dp"
+                    android:layout_marginTop="24dp"
+                    android:text="@string/buy_total_title"
+                    android:textAppearance="@style/UiKit.TextAppearance.Regular.Text3"
+                    android:textColor="@color/text_night"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toBottomOf="@id/recyclerViewMethods" />
+
+                <TextView
+                    android:id="@+id/textViewTotal"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_new_buy.xml:128]: textViewTotalValue

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/showKeyboard.kt (hunk 3 above)
Comment (gslevinkov) [reply ui-kit/src/main/kotlin/org/p2p/uikit/utils/showKeyboard.kt:21]: KeyboardExt.kt

**Review by gslevinkov — APPROVED** [review]

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/showKeyboard.kt (hunk 3 above)
Comment (davran1001) [reply ui-kit/src/main/kotlin/org/p2p/uikit/utils/showKeyboard.kt:21]: OOPS, i didn't see that ))

**Pull request merged**